### PR TITLE
Use client authentication for authentication context internal API calls instead of Mutual SSL

### PR DESCRIPTION
**Pull request by deshankoswatte**
### Proposed changes in this pull request
- This pull request makes sure that client authentication for authentication context internal API calls instead of Mutual SSL.
- Fixes: https://github.com/wso2/product-is/issues/11251.

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [ ] **Is the PR labeled correctly?**

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components.
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [ ] **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] **Are all UI and API inputs run through forms or serializers?**
- [ ] **Are all external inputs validated and sanitized appropriately?**
- [ ] **Does all branching logic have a default case?**
- [ ] **Does this solution handle outliers and edge cases gracefully?**
- [ ] **Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes should be documented.
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented.
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

File: features/authentication-framework/org.wso2.carbon.identity.application.authentication.framework.server.feature/resources/EndpointConfig.properties.j2 [hunk 1]
```diff
@@ -35,3 +35,5 @@ tls.protocol={{identity.auth_framework.endpoint.tls_protocol}}
 app.name={{identity.auth_framework.endpoint.app_name}}
 app.password={{identity.auth_framework.endpoint.app_password}}
 identity.server.origin={{identity.auth_framework.identity_server_origin}}
+basic.auth.name={{identity.auth_framework.endpoint.basic_auth_name}}
+basic.auth.password={{identity.auth_framework.endpoint.basic_auth_password}}
```
Comment (rksk) [inline features/authentication-framework/org.wso2.carbon.identity.application.authentication.framework.server.feature/resources/EndpointConfig.properties.j2:39]:
```suggestion
identity.server.service.access.username={{identity.auth_framework.endpoint.username}}
identity.server.service.access.password={{identity.auth_framework.endpoint.password}}
```

File: features/identity-mgt/org.wso2.carbon.identity.mgt.server.feature/resources/RecoveryEndpointConfig.properties.j2 [hunk 2]
```diff
@@ -24,3 +24,5 @@ app.password={{identity.auth_framework.endpoint.app_password}}
 identity.server.service.access.username={{identity.auth_framework.endpoint.username}}
 identity.server.service.access.password={{identity.auth_framework.endpoint.password}}
 secretProvider={{identity.auth_framework.endpoint.securevault_secret_handler}}
+basic.auth.name={{identity.auth_framework.endpoint.basic_auth_name}}
+basic.auth.password={{identity.auth_framework.endpoint.basic_auth_password}}
```
Comment (rksk) [inline features/identity-mgt/org.wso2.carbon.identity.mgt.server.feature/resources/RecoveryEndpointConfig.properties.j2:28]:
```suggestion
```

File: features/identity-mgt/org.wso2.carbon.identity.mgt.server.feature/resources/RecoveryEndpointConfig.properties [hunk 3]
```diff
@@ -18,3 +18,5 @@
 #identity.server.service.contextURL=https://localhost:9443/services/
 app.name=dashboard
 app.password=dashboard
+basic.auth.name=admin
+basic.auth.password=admin
```
Comment (rksk) [inline features/identity-mgt/org.wso2.carbon.identity.mgt.server.feature/resources/RecoveryEndpointConfig.properties:22]:
```suggestion
```

File: features/authentication-framework/org.wso2.carbon.identity.application.authentication.framework.server.feature/resources/EndpointConfig.properties [hunk 4]
```diff
@@ -31,3 +31,5 @@ tls.protocol=TLSv1.2
 app.name=dashboard
 app.password=dashboard
 identity.server.origin=${carbon.protocol}://${carbon.host}:${carbon.management.port}
+basic.auth.name=admin
+basic.auth.password=admin
```
Comment (rksk) [inline features/authentication-framework/org.wso2.carbon.identity.application.authentication.framework.server.feature/resources/EndpointConfig.properties:35]:
```suggestion
identity.server.service.access.username=admin
identity.server.service.access.password=admin
```

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint.util/src/main/java/org/wso2/carbon/identity/application/authentication/endpoint/util/Constants.java [hunk 5]
```diff
@@ -58,6 +58,8 @@ public class Constants {
     public static final String CONFIG_APP_NAME = "app.name";
     public static final String CONFIG_APP_PASSWORD = "app.password";
     public static final String CONFIG_SERVER_ORIGIN = "identity.server.origin";
+    public static final String CONFIG_BASIC_AUTH_NAME = "basic.auth.name";
+    public static final String CONFIG_BASIC_AUTH_PASSWORD = "basic.auth.password";
```
Comment (rksk) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint.util/src/main/java/org/wso2/carbon/identity/application/authentication/endpoint/util/Constants.java:62]:
```suggestion
    public static final String CONFIG_BASIC_AUTH_NAME = "identity.server.service.access.username";
    public static final String CONFIG_BASIC_AUTH_PASSWORD = "identity.server.service.access.password";
```

File: components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint.util/src/main/java/org/wso2/carbon/identity/mgt/endpoint/util/IdentityManagementEndpointConstants.java [hunk 6]
```diff
@@ -36,6 +36,8 @@ public static final class ServiceConfigConstants {
         public static final String SERVICE_CONTEXT_URL = "identity.server.service.contextURL";
         public static final String APP_NAME = "app.name";
         public static final String APP_PASSWORD = "app.password";
+        public static final String BASIC_AUTH_NAME = "basic.auth.name";
+        public static final String BASIC_AUTH_PASSWORD = "basic.auth.password";
```
Comment (rksk) [inline components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint.util/src/main/java/org/wso2/carbon/identity/mgt/endpoint/util/IdentityManagementEndpointConstants.java:40]:
```suggestion
        public static final String BASIC_AUTH_NAME = "identity.server.service.access.username";
        public static final String BASIC_AUTH_PASSWORD = "identity.server.service.access.password";
```

**Review by rksk — CHANGES_REQUESTED** [review]

File: components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint.util/src/main/java/org/wso2/carbon/identity/mgt/endpoint/util/IdentityManagementEndpointConstants.java (hunk 6 above)
Comment (deshankoswatte) [reply components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint.util/src/main/java/org/wso2/carbon/identity/mgt/endpoint/util/IdentityManagementEndpointConstants.java]: Fixed with https://github.com/wso2/carbon-identity-framework/pull/3968/commits/87f4579c071f7d64026c1f53842e350b6f532c4e.

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint.util/src/main/java/org/wso2/carbon/identity/application/authentication/endpoint/util/Constants.java (hunk 5 above)
Comment (deshankoswatte) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint.util/src/main/java/org/wso2/carbon/identity/application/authentication/endpoint/util/Constants.java]: Fixed with https://github.com/wso2/carbon-identity-framework/pull/3968/commits/87f4579c071f7d64026c1f53842e350b6f532c4e.

File: features/authentication-framework/org.wso2.carbon.identity.application.authentication.framework.server.feature/resources/EndpointConfig.properties (hunk 4 above)
Comment (deshankoswatte) [reply features/authentication-framework/org.wso2.carbon.identity.application.authentication.framework.server.feature/resources/EndpointConfig.properties]: Fixed with https://github.com/wso2/carbon-identity-framework/pull/3968/commits/87f4579c071f7d64026c1f53842e350b6f532c4e.

File: features/identity-mgt/org.wso2.carbon.identity.mgt.server.feature/resources/RecoveryEndpointConfig.properties (hunk 3 above)
Comment (deshankoswatte) [reply features/identity-mgt/org.wso2.carbon.identity.mgt.server.feature/resources/RecoveryEndpointConfig.properties]: Fixed with https://github.com/wso2/carbon-identity-framework/pull/3968/commits/87f4579c071f7d64026c1f53842e350b6f532c4e.

File: features/identity-mgt/org.wso2.carbon.identity.mgt.server.feature/resources/RecoveryEndpointConfig.properties.j2 (hunk 2 above)
Comment (deshankoswatte) [reply features/identity-mgt/org.wso2.carbon.identity.mgt.server.feature/resources/RecoveryEndpointConfig.properties.j2]: Fixed with https://github.com/wso2/carbon-identity-framework/pull/3968/commits/87f4579c071f7d64026c1f53842e350b6f532c4e.

File: features/authentication-framework/org.wso2.carbon.identity.application.authentication.framework.server.feature/resources/EndpointConfig.properties.j2 (hunk 1 above)
Comment (deshankoswatte) [reply features/authentication-framework/org.wso2.carbon.identity.application.authentication.framework.server.feature/resources/EndpointConfig.properties.j2]: Fixed with https://github.com/wso2/carbon-identity-framework/pull/3968/commits/87f4579c071f7d64026c1f53842e350b6f532c4e.

**Review by rksk — APPROVED** [review]

**Pull request merged**